### PR TITLE
tabulation and print test deck code for either-neither

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: client/src/utils/election.ts [hunk 1]
```diff
@@ -16,6 +22,54 @@ export const getContests = ({
...
+    resultContests.splice(
+      eitherNeitherIndex,
+      1,
+      eitherNeitherContest,
+      pickOneContest
+    )
+  })
+
+  return resultContests
+}
```
Comment (eventualbuddha) [inline client/src/utils/election.ts:71]:
```suggestion
export const expandEitherNeitherContests = (
  contests: Contests
): Exclude<AnyContest, MsEitherNeitherContest>[] =>  contests.flatMap((contest) =>
    contest.type !== 'ms-either-neither'
      ? [contest]
      : [
          {
            type: 'yesno',
            id: contest.eitherNeitherContestId,
            title: `${contest.title} – Either/Neither`,
            districtId: contest.districtId,
            section: contest.section,
            description: contest.description,
            yesOption: contest.eitherOption,
            noOption: contest.neitherOption,
          },
          {
            type: 'yesno',
            id: contest.pickOneContestId,
            title: `${contest.title} – Pick One`,
            districtId: contest.districtId,
            section: contest.section,
            description: contest.description,
            yesOption: contest.firstOption,
            noOption: contest.secondOption,
          },
        ]
  )
```

Functionally the same, except I replaced `--` with `–`. Also made the return type more specific.

File: client/src/utils/election.ts [hunk 2]
```diff
@@ -16,6 +22,54 @@ export const getContests = ({
       ballotStyle.partyId === c.partyId
   )
 
+export const getEitherNeitherContests = (
+  contests: Contests
+): MsEitherNeitherContest[] =>
+  contests
+    .filter((c) => c.type === 'ms-either-neither')
+    .map((c) => c as MsEitherNeitherContest)
```
Comment (eventualbuddha) [inline client/src/utils/election.ts:30]:
```suggestion
  contests.filter(
    (c): c is MsEitherNeitherContest => c.type === 'ms-either-neither'
  )
```

If `filter` is given a type predicate the return value will be inferred correctly.

File: client/src/screens/PrintTestDeckScreen.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ const TestDeckBallots = ({
 
       const numBallots = Math.max(
         ...contests.map((c) =>
-          c.type === 'yesno' ? 2 : (c as CandidateContest).candidates.length
+          c.type === 'candidate' ? (c as CandidateContest).candidates.length : 2
```
Comment (eventualbuddha) [inline client/src/screens/PrintTestDeckScreen.tsx:68]:
```suggestion
          c.type === 'candidate' ? c.candidates.length : 2
```

Unnecessary cast.

**Comment by beausmith** [comment]
@benadida Wanted to point out that I merged a PR into this PR as I needed the fixes in this branch in order o complete the fix.

**Comment by benadida** [comment]
> @benadida Wanted to point out that I merged a PR into this PR as I needed the fixes in this branch in order o complete the fix.

Yep, saw it, makes perfect sense.

**Pull request merged**